api/src/Entity/ContentNode.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+namespace App\Entity;
+
+use ApiPlatform\Core\Annotation\ApiResource;
+use App\Repository\ContentNodeRepository;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity(repositoryClass=ContentNodeRepository::class)
+ */
+#[ApiResource]
+class ContentNode extends BaseEntity {
+    /**
+     * @ORM\OneToOne(targetEntity=Activity::class, inversedBy="rootContentNode", cascade={"persist", "remove"})
+     */
+    private ?Activity $owner = null;
+
+    /**
+     * @ORM\ManyToOne(targetEntity=ContentNode::class, inversedBy="rootDescendants")
+     * TODO make not null, and get fixtures to run
+     * @ORM\JoinColumn(nullable=true)
+     */
+    private ContentNode $root;
+
+    /**
+     * @ORM\OneToMany(targetEntity=ContentNode::class, mappedBy="root")
+     */
+    private Collection $rootDescendants;
+
+    /**
+     * @ORM\ManyToOne(targetEntity=ContentNode::class, inversedBy="children")
+     */
+    private ?ContentNode $parent = null;
+
+    /**
+     * @ORM\OneToMany(targetEntity=ContentNode::class, mappedBy="parent")
+     */
+    private Collection $children;
+
+    /**
+     * @ORM\Column(type="text", nullable=true)
+     */
+    private ?string $slot = null;
+
+    /**
+     * @ORM\Column(type="integer", nullable=true)
+     */
+    private ?int $position = null;
+
+    /**
+     * @ORM\Column(type="json", nullable=true)
+     */
+    private array $jsonConfig = [];
+
+    /**
+     * @ORM\Column(type="text", nullable=true)
+     */
+    private ?string $instanceName = null;
+
+    /**
+     * @ORM\ManyToOne(targetEntity=ContentType::class)
+     * @ORM\JoinColumn(nullable=false)
+     */
+    private ?ContentType $contentType = null;
+
+    public function __construct() {
+        $this->rootDescendants = new ArrayCollection();
```

Comment:
hier wurde vorher $this->root = $this gesetzt

Reply:
Wird durch `$this->addRootDescendant($this)` erledigt.